ros/angel_system_nodes/angel_system_nodes/task_monitor_v2.py

```diff
@@ -168,10 +188,22 @@ def publish_task_state_message(self):
 
         # Populate steps and current step
         with self._hmm_lock:
-            message.steps = self._hmm.model.class_str
-            message.current_step_id = self._hmm.model.class_str.index(self._current_step)
+            message.steps = self._hmm.model.class_str[1:] # exclude background
 
-        message.current_step = self._current_step
+            if self._current_step is None:
+                message.current_step_id = 0
+            else:
+                hmm_current_step_id = message.steps.index(
+                    self._current_step
+                )
+                message.current_step_id = hmm_current_step_id + 1
+
+        if message.current_step_id == len(message.steps):
+            # Because we don't have a done step, keep this previous step but
+            # advance the index so the ARUI can see we're done
+            message.current_step = message.steps[message.current_step_id - 1]
```

Comment:
Is this describing the mechanic in which we signify the task is "done"? If so, I think it would be more understandable to be more explicit about this and use the `task_complete_confidence` field we added (set to `1.0` here).